### PR TITLE
implement IPC cert and key generation

### PR DESCRIPTION
**Pull request by misteriaud**

### CODE REVIEW

**Comment by misteriaud** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: pkg/api/security/cert/cert_getter.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+// Requires that the config has been set up before calling
+func CreateOrFetchAgentIPCCert(config configModel.Reader) ([]byte, []byte, error) {
+	return fetchAgentIPCCert(config, true)
+}
+
+func fetchAgentIPCCert(config configModel.Reader, certCreationAllowed bool) ([]byte, []byte, error) {
+	certPath := GetCertFilepath(config)
+
+	// Create cert&key if it doesn't exist and if permitted by calling func
+	if _, e := os.Stat(certPath + ".cert"); os.IsNotExist(e) && certCreationAllowed {
```
Comment (sgnn7) [inline pkg/api/security/cert/cert_getter.go:54]: drive-by nit: Certs are usually `.crt` extension (instead of `.cert`)
